libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -253,13 +289,28 @@ protected override async Task<bool> OnPostBubbleEventAsync(DialogContext dc, Dia
             return await ProcessEventAsync(sequenceContext, dialogEvent, preBubble: false, cancellationToken: cancellationToken).ConfigureAwait(false);
         }
 
-        protected async Task<bool> ProcessEventAsync(SequenceContext sequenceContext, DialogEvent dialogEvent, bool preBubble, CancellationToken cancellationToken = default)
+        protected virtual async Task<bool> ProcessEventAsync(SequenceContext sequenceContext, DialogEvent dialogEvent, bool preBubble, CancellationToken cancellationToken = default(CancellationToken))
         {
             // Save into turn
             sequenceContext.GetState().SetValue(TurnPath.DIALOGEVENT, dialogEvent);
 
             EnsureDependenciesInstalled();
 
+            // Count of events processed
+            var count = sequenceContext.GetState().GetValue<uint>(DialogPath.EventCounter);
+            sequenceContext.GetState().SetValue(DialogPath.EventCounter, ++count);
+
+            // Save schema information
+            if (this.Schema != null)
+            {
+                sequenceContext.GetState().SetValue(TurnPath.SCHEMA, this.Schema.Schema);
```

Comment:
>TurnPath.SCHEMA, [](start = 52, length = 16)

You don't need to do this.  Current Dialog class is available via "instance" memory scope, so "instance.schema" should give you the instance class data for the active dialog.